### PR TITLE
Core with no javax

### PR DESCRIPTION
**Pull request by jad-elkhoury**
## Description

Remove the dependency on Javax.Servlet from Core. We migrate the Jax-rs Providers from OSLC4J Core to Server.

See following branch of refImpl QM to test in action.
https://github.com/oslc-op/refimpl/compare/master...CoreWithNoJavax

## ToDo

- [ ] Reinstate the removed packages/modules, tagging them as Deprecated. Currently removed to ensure the new packages/modules work with no dependencies on the old ones.

## Checklist

- [X] This PR adds an entry to the CHANGELOG. _See https://keepachangelog.com/en/1.0.0/ for instructions. Minor edits are exempt._
- [x] This PR was tested on at least one Lyo OSLC server or adds unit/integration tests.
- [ ] This PR does NOT break the API

### CODE REVIEW

**Review by berezovskyi — CHANGES_REQUESTED** [review]
Thank you, very happy with the result!

It would have been better to see the old classes `@Deprecated` in the same pull request, as I am a bit worried if we will miss anything when you will bring the old code back.

Added a few comments, some of them can be filed as future issues or incorporated into the PR that restores the old code.

I also made a PR to annotate some/most of the new code with `@ApiStatus.Experimental` so that we can make more breaking changes to these new files in Lyo 5.2 etc: https://github.com/eclipse/lyo/pull/283. This is a common practice for large/mature OSS projects:



If you agree with that, let's merge PR #283 into #282 (this one) first, and then #282 (this one) into `master`.

Finally, I think we should bump the version to `5.1.0-SNAPSHOT` as we now include new things instead of just bugfixes in the release.

Again, thank you for the PR!

**Comment by jadelkhoury** [comment]
I should have marked the PR as draft. Yes, we should keep/restore the old classes before we merge. Thought it was worth seeing these new classes foest

**Comment by jad-elkhoury** [comment]
@berezovskyi A thought about deprecating: The aim of this migration is to rid Core from Javax.servlet.
Is there a way we can put the deprecated classes in another project than Core, yet keep their package names?
It would be nice to be able to have a Core without javax.servlet.

**Comment by berezovskyi** [comment]
Yes, we can. We may want to make core depend on the new artifacts until 6.0 but we should be able to `exclude` them when using now.

File: core/oslc4j-core/pom.xml [hunk 1]
```diff
@@ -70,6 +65,17 @@
       <artifactId>jakarta.xml.bind-api</artifactId>
     </dependency>
 
+    <dependency>
+        <groupId>org.apache.wink</groupId>
```
[the remark this answers is not in the document]
Comment (jad-elkhoury) [reply core/oslc4j-core/pom.xml:69]: please check. fixed in last commit

File: core/oslc-trs/src/test/java/org/eclipse/lyo/core/trs/TrackedResourceSetTest.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 import org.apache.jena.rdf.model.Model;
 import java.net.URI;
 import java.net.URISyntaxException;
-import org.eclipse.lyo.oslc4j.provider.jena.JenaModelHelper;
+import org.eclipse.lyo.oslc4j.core.jena.JenaModelHelper;
```
[the remark this answers is not in the document]
Comment (jad-elkhoury) [reply core/oslc-trs/src/test/java/org/eclipse/lyo/core/trs/TrackedResourceSetTest.java:20]: yes

File: core/oslc4j-core/src/main/java/org/eclipse/lyo/oslc4j/core/OSLC4JUtils.java [hunk 3]
```diff
@@ -246,59 +245,6 @@ public static String getServletPath()
 		return servletPath;
 	}
 
-	/**
```
[the remark this answers is not in the document]
Comment (jad-elkhoury) [reply core/oslc4j-core/src/main/java/org/eclipse/lyo/oslc4j/core/OSLC4JUtils.java:249]: will bring bak

File: server/provider-utils/pom.xml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        <dependency>
+            <groupId>jakarta.ws.rs</groupId>
+            <artifactId>jakarta.ws.rs-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>io.github.classgraph</groupId>
+            <artifactId>classgraph</artifactId>
+            <version>4.8.143</version>
```
[the remark this answers is not in the document]
Comment (jad-elkhoury) [reply server/provider-utils/pom.xml]: please check. fixed in last commit

File: server/provider-utils/pom.xml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.eclipse.lyo.oslc4j.server</groupId>
+        <artifactId>lyo-server-build</artifactId>
+        <version>5.0.1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <groupId>org.eclipse.lyo.server</groupId>
+    <artifactId>provider-utils</artifactId>
+    <name>Lyo :: Core :: Provider Utils</name>
```
[the remark this answers is not in the document]
Comment (jad-elkhoury) [reply server/provider-utils/pom.xml]: please check. fixed in last commit

**Comment by jad-elkhoury** [comment]
> Yes, we can. We may want to make core depend on the new artifacts until 6.0 but we should be able to `exclude` them when using now.

You might need to show me what you mean. But good to know it is possible.

File: core/oslc4j-core/pom.xml [hunk 6]
```diff
@@ -70,6 +65,15 @@
...
+        <artifactId>wink-json4j</artifactId>
+    </dependency>
```
Comment (berezovskyi) [inline core/oslc4j-core/pom.xml:71]: we should not bring Wink back into non-legacy Lyo modules

Comment (jad-elkhoury) [reply core/oslc4j-core/pom.xml:71]: This Wink has always been there.

**Comment by jadelkhoury** [comment]
Should close this PR If/when the superceeding one is merged
https://github.com/eclipse/lyo/pull/284

**Pull request closed without merging**